### PR TITLE
Ots16 samples function

### PR DESCRIPTION
**Pull request by liz-fw**
- refactors logic for status code and run type IDs (creates new script for these functions)
- creates function for querying on status code/run assignment to return list of samples needing further analysis
- #9 
- #40

### CODE REVIEW

File: R/assign_codes.R [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#' * needs_ots_16
+#' * repeat_ots_28
+#' * repeat_ots_16
+#' * potential_heterozygotes
+#'
+#' @export
+get_samples_needing_action <- function(con) {
+  needs_ots_16 <- dplyr::tbl(con, "sample_status") |>
+    collect() |>
+    filter(status_code_id == 8) |>
```
Comment (ergz) [inline R/assign_codes.R:87]: * add `dplyr::` to all dplyr functions
* move filter, joins and other db operations before calling collect()
* create a main `dplyr::tbl(con, "sample_status")` at the top of the function that each of these sub-queries can use

File: R/get-sample.R [hunk 2]
```diff
@@ -9,3 +9,42 @@ get_samples <- function(con, ...) {
...
+
+#' Generate Subsample
+#' @description `get_subsample()` selects sample IDs from the field for assay workflow
+#' @param con
+#' @param sample_event_id
+#' @examples
+#' @export
+#' @md
+
+get_subsample <- function(con, sample_event_id) {
```
Comment (ergz) [inline R/get-sample.R:21]: remove this for now since you have a branch just for tackling this issue

**Comment by liz-fw** [comment]
@ergz I made the changes you suggested and moved the function to the sample-status.R file (I figured that made more sense). It looks like we already have a function get_sample_status() where you can pass sample IDs and get their status in that file

**Review by ergz — APPROVED** [review]
looks good!

**Pull request merged**